src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs

```diff
@@ -47,16 +47,21 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 
         var thunks = endpoints.Select((endpoint, _) => $$"""
             [{{StaticRouteHandlerModelEmitter.EmitSourceKey(endpoint)}}] = (
-           (del, builder) =>
+           (methodInfo, options) =>
             {
-                builder.Metadata.Add(new SourceKey{{StaticRouteHandlerModelEmitter.EmitSourceKey(endpoint)}});
+                if (options == null)
+                {
+                    return new RequestDelegateMetadataResult { EndpointMetadata = new List<object>().AsReadOnly() };
```

Comment:
nit:
Replace `new List<object>().AsReadOnly()` with `ReadOnlyCollection<object>.Empty`, at least 3 places can be changed

Reply:
TIL! Looks like this was added in 8.0? Very nice...

Reply:
Oh, I just assumed it existed and saw that it did 😄 
Didn't realize it's new!

Reply:
I made the same comment, but you can use `Array.Empty<object>()` instead.

Reply:
I ended up using `ReadOnlyCollection<object>`. In the interest of readability it does a better job of communicating the type passed to the initializer here and I don't think there's any memory implications with that singleton.